butane_core/src/fkey.rs

```diff
@@ -206,9 +206,14 @@ where
 }
 
 #[cfg(feature = "fake")]
-/// Fake data support is currently limited to empty ForeignKey relationships.
-impl<T: DataObject> Dummy<Faker> for ForeignKey<T> {
-    fn dummy_with_rng<R: rand::Rng + ?Sized>(_: &Faker, _rng: &mut R) -> Self {
-        Self::new_raw()
+/// A dummy value for T will be created, however it also has to have the Clone
+/// trait in order for the mutable value to be obtained so that it can be saved.
+impl<T: DataObject + Dummy<Faker>> Dummy<Faker> for ForeignKey<T> {
```

Comment:
If we proceed with this, and `.clone()` is needed to get the `T` to be saved, IMO we should do:
```diff
-impl<T: DataObject + Dummy<Faker>>...
+impl<T: Clone + DataObject + Dummy<Faker>>...
``` 
and then document how to save the fake data.